internal/linker/linker.go

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) 2022, The Garble Authors.
+// See LICENSE for licensing information.
+
+package linker
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"embed"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"github.com/bluekeyes/go-gitdiff/gitdiff"
+	"github.com/gofrs/flock"
+	"io"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+const (
+	MagicValueEnv = "GARBLE_LNK_MAGIC"
+
+	cacheDirName   = ".garble"
```

Comment:
directories inside UserCacheDir should not be "dot files", e.g. `~/.config` has `vlc` or `nvim` in my machine.